app/connectors/EmailConnector.scala

```diff
@@ -16,29 +16,24 @@
 
 package connectors
 
-import scala.concurrent.ExecutionContext
-import scala.concurrent.Future
+import com.google.inject.Inject
 import models.SendEmailRequest
-import play.api.{Configuration, Play}
+import play.api.{Configuration, Environment}
 import play.api.Mode.Mode
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
 import uk.gov.hmrc.play.config.ServicesConfig
-import utils.WSHttp
-import uk.gov.hmrc.http.{HeaderCarrier, HttpPost, HttpResponse}
 
-object EmailConnector extends EmailConnector with ServicesConfig {
-  override protected def mode: Mode = Play.current.mode
-  override protected def runModeConfiguration: Configuration = Play.current.configuration
+import scala.concurrent.{ExecutionContext, Future}
 
-  override val httpPost = WSHttp
-  override val emailUrl = baseUrl("email")
-}
+class EmailConnector @Inject()(val runModeConfiguration: Configuration, environment: Environment, http: HttpClient) extends ServicesConfig {
+
+  override def mode: Mode = environment.mode
 
-trait EmailConnector {
+  val emailUrl = baseUrl("email")
 
-  val httpPost: HttpPost
-  val emailUrl: String
   def url(path: String) = s"$emailUrl$path"
 
   def sendEmail(sendEmailRequest: SendEmailRequest)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] =
```

Comment:
you may want to give the class an execution context, rather than methods

Reply:
done